packages/core/core-v2/src/Parcel.js

```diff
@@ -0,0 +1,191 @@
+// @flow
+'use strict';
+import path from 'path';
+import { AbortController } from 'abortcontroller-polyfill/dist/cjs-ponyfill';
+import Watcher from '@parcel/watcher';
+import PQueue from 'p-queue';
+import AssetGraph from './AssetGraph';
+import { Node } from './Graph';
+import type { Dependency, Asset, File } from './types';
+import TransformerRunner from './TransformerRunner';
+import ResolverRunner from './ResolverRunner';
+import BundlerRunner from './BundlerRunner';
+import PackagerRunner from './PackagerRunner';
+
+// TODO: use custom config if present
+const defaultConfig = require('@parcel/config-default');
+
+const AbortError = new Error('Aborted!');
```

Comment:
Agreed it could be more descriptive, though I don't think this error will ever get logged. I should probably add a catch in the build function that logs out an error if there was an actual failure to build and just ignore if the rejection is because the build was aborted. I haven't spent a lot of time working outside of the happy path yet :P